src/lexers/jsx-lexer.js

```diff
@@ -112,9 +115,35 @@ export default class JsxLexer extends JavascriptLexer {
         }
       }
       else if (child.type === 'JSXExpressionContainer') {
+        // strip empty expressions
+        if (child.expression.type === 'JSXEmptyExpression')
+          return {
+            type: 'text',
+            content: ''
+          }
+
+        // strip properties from ObjectExpressions
+        // annoying (and who knows how many other exceptions we'll need to write) but necessary
+        else if (child.expression.type === 'ObjectExpression') {
+          let content = '{'
+          let start = child.expression.start
+
+          child.expression.properties.forEach(prop => {
+            content += originalString.slice(start, prop.key.end)
+            start = prop.value.end
+          })
+          content += originalString.slice(start, child.expression.end) + '}'
+
+          return {
+            type: 'js',
+            content
+          }
+        }
+
+        // slice on the expression so that we ignore comments around it
         return {
           type: 'js',
-          content: originalString.slice(child.start, child.end)
+          content: `{${originalString.slice(child.expression.start, child.expression.end)}}`
```

Comment:
We're slicing on the `.expression` b/c if we do that it very neatly strips comments out.